designs/0031-deprecation-schedule.md

```diff
@@ -0,0 +1,254 @@
+request, if they can foresee code improvements which can occur at that time.
+
+## Code reviewers
+From a code reviewer standpoint, it is important to ask the following questions:
+1. Is this deprecation tied to a useful feature or noticeable improvement to the
+   underyling code (not just 'for the sake of it')?
+2. Does the new warning meet the criteria above and explain itself clearly?
+3. Is there a simple (preferably automated) way for users to accommodate this
+   change, e.g. through the `print-canonical` flag to stanc3 or a
+   find-and-replace feature of their text editor?
+4. Is the removal timeline appropriate for this change?
+
+## Releases
+Finally, during the release cycle an additional bullet point should be added to
+the release checklist to ensure that the deprecations slated for removal in the
+coming version have been properly handled. To accommodate this, the documentation
+page on deprecations should be sorted/categorized by removal version.
+
+Additionally, the release notes should contain a section on **Deprecations** and
+**Expired Deprecations** (alt. **Removals**) for each version. This should
+duplicate the information on the what and how of the change from the error
+message and documentation.
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Drawbacks to this are rather obvious - backwards compatibility is certainly a
+virtue, and any breaking change will lead to some headaches for a portion of the
+user base, no matter how small. In the limit, we certainly want to avoid any
+large enough change that users prefer not to update (a core reason these changes
+should be tied to improvements in the language, not just for their own sake) and
+any changes which require dedicated rewriting or rethinking of models. This
+desire motivated the definition of 'minor change' above and the restriction to
+primarily automatable syntax changes.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+This design allows for us to care for the long term health of the Stan language
+and codebase while clearly communicating our intent to users. Small changes can
+be made to make the development of new features easier without "sneaking up on"
+anyone or breaking things willy-nilly.
+
+There are two main alternatives to this proposal:
+1. The first is to stick strictly to the semantic versioning standards, and make
+   a strong push toward a Stan version 3.0 which makes backwards incompatible
+   changes.
+
+   Unless we attempt to systematically release major versions with some
+   regularity, we still end up with the same problem of these tiny issues
+   accumulating in the meantime. Additionally, saving all such minor changes for
+   a major version bump dramatically increases the friction for updating. While
+   some may find it preferable to have to update their models only once, the
+   small requirement of (at most) 1-2 changes per version is likely preferable
```

Comment:
I wouldn't legislate the number of deprecations per release.  For one things, that would require a way of counting them!

Reply:
We will be counting them when we produce release notes, but I think you're correct its hard to codify this fully.